azurerm/resource_arm_api_management_api.go

```diff
@@ -454,3 +477,30 @@ func flattenApiManagementApiSubscriptionKeyParamNames(paramNames *apimanagement.
 
 	return []interface{}{result}
 }
+
+func expandApiManagementAuthenticationSettings(input []interface{}) *apimanagement.AuthenticationSettingsContract {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
```

Comment:
This caused a test to panic so we should confirm that `input[0] != nil` before casting this to a `map[string]interface{}`

```

------- Stdout: -------
=== RUN   TestAccAzureRMApiManagementApi_importUpdate

------- Stderr: -------
2019/11/15 17:21:04 [DEBUG] Registering Data Sources for "Compute"..
2019/11/15 17:21:04 [DEBUG] Registering Resources for "Compute"..
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1247 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm.expandApiManagementAuthenticationSettings(...)
	/opt/teamcity-agent/work/458e5e4800bd94f6/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_api_management_api.go:486
github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmApiManagementApiCreateUpdate(0xc0002c9490, 0x3b50f20, 0xc0014fe000, 0x0, 0x0)
```